index.js

```diff
@@ -9,11 +9,13 @@ async function basicPlugin (fastify, opts) {
     throw new Error('Basic Auth: Missing validate function')
   }
   const authenticateHeader = getAuthenticateHeader(opts.authenticate)
+  const header = opts.header || 'authorization'
```

Comment:
```suggestion
  const header = opts.header.toLowerCase() || 'authorization'
```

Reply:
Added a check for the existence of `opts.header`